grammars/ruby.cson

```diff
@@ -293,11 +333,32 @@
       '3':
         'name': 'entity.name.function.ruby'
     'comment': ' the optional name is just to catch the def also without a method-name'
-    'match': '(?x)\n\t\t\t         (?=def\\b)                                                           # an optimization to help Oniguruma fail fast\n\t\t\t         (?<=^|\\s)(def)\\b                                                    # the def keyword\n\t\t\t         ( \\s+                                                               # an optional group of whitespace followed by…\n\t\t\t           ( (?>[a-zA-Z_]\\w*(?>\\.|::))?                                      # a method name prefix\n\t\t\t             (?>[a-zA-Z_]\\w*(?>[?!]|=(?!>))?                                 # the method name\n\t\t\t             |===?|!=|>[>=]?|<=>|<[<=]?|[%&`/\\|]|\\*\\*?|=?~|[-+]@?|\\[]=?) ) )?  # …or an operator method\n\t\t\t        '
+    'match': '''(?x)
+      (?=def\\b)                            # optimization to help Oniguruma fail fast
+      (?<=^|\\s)(def)\\b
+      (
+        \\s+
+        (
+          (?>[a-zA-Z_]\\w*(?>\\.|::))?      # method prefix
+          (?>                               # method name
+            [a-zA-Z_]\\w*(?>[?!]|=(?!>))?
+            |===?|!=|>[>=]?|<=>|<[<=]?|[%&`/\\|]|\\*\\*?|=?~|[-+]@?|\\[]=?
+          )
+        )
+      )?
+    '''
     'name': 'meta.function.method.without-arguments.ruby'
   }
   {
-    'match': '\\b(0[xX]\\h(?>_?\\h)*|\\d(?>_?\\d)*(\\.(?![^[:space:][:digit:]])(?>_?\\d)*)?([eE][-+]?\\d(?>_?\\d)*)?|0[bB][01]+)\\b'
+    'match': '''(?x)
+      \\b(
+        \\d(?>_?\\d)*                               # 100_000
+        (\\.(?![^[:space:][:digit:]])(?>_?\\d)*)?   # fractional part
+        ([eE][-+]?\\d(?>_?\\d)*)?                   # 1.23e-4
+      | 0[xX]\\h(?>_?\\h)*
```

Comment:
I don't know, shouldn't they (`decimal|hex|binary`) be on the same level?

![image](https://cloud.githubusercontent.com/assets/18070858/16892674/a925347c-4b4d-11e6-9ba0-7e85d0d44ac3.png)
![image](https://cloud.githubusercontent.com/assets/18070858/16892676/cdfb269e-4b4d-11e6-93cd-bcce014554cf.png)
![image](https://cloud.githubusercontent.com/assets/18070858/16892702/f2e98e4a-4b4e-11e6-991c-66f045dae705.png)

Pick one, then I'll stick to it.